src/transformStateWithClones.js

```
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const log = [];
+  let copiedState = structuredClone(state);
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        copiedState = {};
+        break;
+      case 'addProperties':
+        Object.assign(copiedState, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete copiedState[key];
+        }
+        break;
+      default:
```

This `default` case meets the requirement of being present, but it doesn't perform any error handling as suggested by the checklist. To make the function more robust, it's good practice to handle unexpected action types explicitly, for example by throwing an error. This prevents silent failures if an invalid action is provided.